web-app/js/plugin/BoxPlot.js

```diff
@@ -370,5 +370,43 @@ BoxPlotView.prototype.submit_job = function (form) {
 
 }
 
+BoxPlotView.prototype.clear_high_dimensional_input = function (div) {
+    RmodulesView.prototype.clear_high_dimensional_input.call(this, div);
+
+    // Only clear bins for corresponding variable (dependent/independent)
+    var variableToBeBinned = document.getElementById("selBinVariableSelection").value
+    if (div == "divDependentVariable" && variableToBeBinned == "DEP" ||
```

Comment:
I think the boxes should be cleared either way, even if binning is disabled. If the user enables binning after clearing, you don't want to see the previously cleared categories, right?